internal/characters/yaoyao/asc.go

```diff
@@ -0,0 +1,72 @@
+package yaoyao
+
+import (
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/glog"
+	"github.com/genshinsim/gcsim/pkg/core/player"
+)
+
+const a4Status = "yaoyao-a4"
+
+func (c *char) a1Ticker() {
+	if !c.StatusIsActive(burstKey) {
+		return
+	}
+	switch c.Core.Player.CurrentState() {
+	case action.DashState, action.JumpState:
+		c.Core.Log.NewEvent("yaoyao a1 triggered", glog.LogCharacterEvent, c.Index).
+			Write("state", c.Core.Player.CurrentState())
+		c.a1Throw()
+	}
+	c.QueueCharTask(c.a1Ticker, 0.6*60)
+}
+
+func (c *char) a1Throw() {
+
+	a1aoe := combat.NewCircleHitOnTarget(c.Core.Combat.Player().Pos(), nil, skillTargetingRad)
+	enemy := c.Core.Combat.RandomEnemyWithinArea(a1aoe, nil)
+	if enemy == nil {
+		return
+	}
+	target := enemy.Pos()
+
+	radishExplodeAoE := combat.NewCircleHitOnTarget(target, nil, radishRad)
+
+	ai := c.burstRadishAI
+	snap := c.Snapshot(&ai)
+	hi := c.getBurstHealInfo(&snap)
+
+	c.Core.QueueAttack(
+		ai,
+		radishExplodeAoE,
+		0,
+		travelDelay,
+		c.makeHealCB(radishExplodeAoE, hi),
+		c.makeC2CB(),
+	)
+}
```

Comment:
This is also subject to the same mechanic as used in `yuegui.go`'s `throw()`.

Reply:
why? `a1Ticker` checks if burst active

Reply:
It could expire for the very last radish.

Reply:
where do i get the skill radish if, for example, i use the burst only?

Reply:
Put the skillRadishAI into a char variable on Init? Could also move burstRadishAI there.

Reply:
Alternatively you could put it in a function just like the heal info.